internal/selfdescribe/monitors.go

```diff
@@ -42,41 +53,54 @@ func monitorsStructMetadata() []monitorMetadata {
 	// Set to track undocumented monitors
 	monTypesSeen := make(map[string]bool)
 
-	filepath.Walk("internal/monitors", func(path string, info os.FileInfo, err error) error {
-		if !info.IsDir() || err != nil {
+	if err := filepath.Walk("internal/monitors", func(path string, info os.FileInfo, err error) error {
+		if err != nil {
 			return err
 		}
-		pkgDoc := packageDoc(path)
-		if pkgDoc == nil {
+
+		if info.IsDir() || info.Name() != monitorMetadataFile {
 			return nil
 		}
-		for monType, monDoc := range monitorDocsInPackage(pkgDoc) {
+
+		var monitorDocs []monitorDocMetadata
+
+		if bytes, err := ioutil.ReadFile(path); err != nil {
+			return errors.Errorf("Unable to read metadata file %s: %s", path, err)
+		} else if err := yaml.Unmarshal(bytes, &monitorDocs); err != nil {
+			return err
+		}
+
+		for _, monitor := range monitorDocs {
+			monType := monitor.MonitorType
+
 			if _, ok := monitors.ConfigTemplates[monType]; !ok {
 				log.Errorf("Found MONITOR doc for monitor type %s but it doesn't appear to be registered", monType)
```

Comment:
Fix up this error message to make it clear the doc is in a YAML file and output the yaml file path where the unregistered monitor appears.